### ISSUE TITLE
Pyright marking attribute type defined as `method = OtherClass.method` from imported module as Unknown

### ISSUE THREAD

**Issue by StockerMC** [microsoft/pyright#2345]
**Describe the bug**
Pyright is marking the type of an attribute (which is a method) defined in a `TYPE_CHECKING` block in a class as `Unknown`. This does not affect non-method attributes (i.e. `attr: <type>` instead of `attr = OtherClass.method`).

This does not happen if the file's parent directory is in the same folder as the `discord` module folder (when the library is cloned), or if the file is in the `discord` module folder. An example file structure where this doesn't occur:
```
discord.py/
    examples/
        subfolder/
            file.py <- if the code was in this file, `discord.Member.create_dm` would have the correct type
    discord/
```

**To Reproduce**
1. `pip install git+https://github.com/Rapptz/discord.py`
2. Paste the code below into a file
3. Run `pyright <file>.py`

**Expected behavior**
The type of `create_dm` of both the `discord.Member` and `Member` class should be the same, and not `Unknown`.

**Screenshots or Code**
```python
from typing import TYPE_CHECKING, Optional
import discord

reveal_type(discord.Member.create_dm) # Unknown
reveal_type(discord.Member.avatar) # Asset | None

class Member:
    # this is in discord.Member
    if TYPE_CHECKING:
        create_dm = discord.User.create_dm
        avatar: Optional[discord.Asset]

reveal_type(Member.create_dm) # (self: User) -> Coroutine[Any, Any, DMChannel]
reveal_type(Member.avatar) # Asset | None
```

**VS Code extension or command-line**
This affects pyright version 1.1.171 and pylance (vscode extension) version 2021.9.3

**Additional context**
The line in discord.py where this is defined is https://github.com/Rapptz/discord.py/blob/master/discord/member.py#L280

**Comment by erictraut** [comment]
This package is marked as "py.typed", so pyright applies the rules specified in [this document](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md). In particular, it assumes that all symbols within a py.typed package will have proper type annotations and treats unannotated symbol as an "unknown" type. It does not fall back on type inference for symbols in a py.typed library for the reasons explained in the document.

In this particular case, the `create_dm` variable does not have a type annotation. If you add one, pyright will honor it.

**Comment by erictraut** [comment]
The command-line version of pyright has an option called "--verifytypes" that you can use to find any public symbols in your py.typed package that are missing annotations.

When I run it on your package, it reports the following:

```
Symbols exported by "discord": 5646
  With known type: 4837
  With partially unknown type: 809
  Functions without docstring: 5580
  Functions without default param: 206
  Classes without docstring: 224

Other symbols referenced but not exported by "discord": 2449
  With known type: 2327
  With partially unknown type: 122

Type completeness score: 85.7%

Completed in 4.097sec
```

And `create_dm` is flagged as not having an annotation:
```
discord.member.Member.create_dm
  /Users/eric/venvs/p39/lib/python3.9/site-packages/discord/member.py:280:9 - error: Type unknown for variable "discord.member.Member.create_dm"
```

**Comment by StockerMC** [comment]
Alright, thanks.
